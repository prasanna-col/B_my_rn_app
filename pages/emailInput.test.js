import React from 'react';
import { TextInput } from 'react-native'
import EmailInput from './emailInput'

// import {shallow} from 'enzyme';
// const Enzyme = require('enzyme');
// const EnzymeAdapter = require('enzyme-adapter-react-15');

// Setup enzyme's react adapter
// Enzyme.configure({ adapter: new EnzymeAdapter() });

// const mock = jest.fn();
// const wrapper = shallow(<App validateEmail={mock} />);
// wrapper.find('TextInput').simulate('changeText', 'abc@yop.in');
// expect(mock).toHaveBeenCalledWith('abc@yop.in');

import { render, fireEvent, cleanup } from '@testing-library/react-native';

// it('does stuff', () => {
//   const mock = jest.fn()
//   const component = render(<EmailInput validateEmail={mock} />)
//   fireEvent.changeText(component.findByType('TextInput'), 'test')
//   expect(mock).toHaveBeenCalledWith('test')
// })

afterEach(cleanup);

describe('<EmailInput />', () => {
  it('should match the snapshot', () => {
    const rendered = render(<EmailInput value={'abacaba'} />).toJSON();
    
    expect(rendered).toMatchSnapshot();
  });
  
  it('should fire onChange events', () => {
    const onChange = jest.fn();
    const rendered = render(<EmailInput onChangeText={onChange} />);
    const inputComponent = rendered.getByTestId('email-input');
    
    fireEvent(inputComponent, 'changeText', 'new text');
    
    // expect(onChange).toHaveBeenCalledWith('new text');
    expect(onChange).toHaveValue("test@mail.com");
  });
  
  it('should fire onSubmit events', () => {
    const onSubmit = jest.fn();
    const rendered = render(<EmailInput validateEmail={onSubmit} />);
    const buttonComponent = rendered.getByTestId('submit-btn');
    
    fireEvent(buttonComponent, 'press');
    
    // expect(onSubmit).toHaveBeenCalledWith('');
    expect(onSubmit).toHaveBeenCalled();
  });
});


// let total = 1 + 1

// it('should run test flawlessly', () => {
//     expect(total).toBe(2)
// })

// it('should fail the test', () => {
//     expect(total).toBe(2)
// })