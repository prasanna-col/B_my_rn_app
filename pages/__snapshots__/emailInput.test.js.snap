// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailInput /> should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      }
    }
  >
    <TextInput
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="#9a73ef"
      style={
        Object {
          "borderColor": "#7a42f4",
          "borderWidth": 1,
          "height": 40,
          "margin": 15,
        }
      }
      testID="email-input"
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "justifyContent": "center",
        "marginBottom": 20,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fdd851",
          "borderRadius": 5,
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "center",
          "marginHorizontal": 20,
          "opacity": 1,
          "width": "90%",
        }
      }
      testID="submit-btn"
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "500",
            "paddingLeft": 10,
          }
        }
      >
        Sumbit
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
